src/plugins/mocap/mocap_impl.cpp

```diff
@@ -73,13 +74,23 @@ Mocap::Result MocapImpl::set_odometry(const Mocap::Odometry& odometry)
 
 bool MocapImpl::send_vision_position_estimate()
 {
+    uint64_t fcu_time_usec{};
+
     _visual_position_estimate_mutex.lock();
     auto vision_position_estimate = _vision_position_estimate;
     _visual_position_estimate_mutex.unlock();
 
     if (!vision_position_estimate.time_usec) {
-        vision_position_estimate.time_usec =
-            static_cast<uint64_t>(_parent->get_time().elapsed_s() * 1e3);
+        fcu_time_usec = std::chrono::duration_cast<std::chrono::microseconds>(
+                            _parent->get_fcu_time().now().time_since_epoch())
+                            .count();
+    } else {
+        fcu_time_usec =
+            std::chrono::duration_cast<std::chrono::microseconds>(
+                _parent->get_fcu_time()
+                    .time_in(std::chrono::microseconds(vision_position_estimate.time_usec))
+                    .time_since_epoch())
```

Comment:
Should we have a helper in `SystemImpl` for this or is this ever only used here?